docs/user/reference/general/initializing.md

```diff
@@ -223,22 +247,136 @@ Glean.initialize(
+
+    @Test
+    fun checkCollectedData() {
+      // The Glean Kotlin SDK testing APIs can be called here.
+    }
+}
+```
+
+This will ensure that metrics are done recording when the other test functions are used.
+
+</div>
+
+<div data-lang="Swift" class="tab">
+
+**Note**: There's no automatic test rule for Glean tests implemented in Swift.
+
+In order to prevent issues with async calls when unit testing the Glean SDK, it is important to put
+the Glean Swift SDK into testing mode. When the Glean Swift SDK is in testing mode,
+it enables uploading and clears the recorded metrics at the beginning of each test run.
+
+Activate it by resetting Glean in your test's setup:
+
+```swift
+// All pings and metrics testing APIs are marked as `internal`
+// so you need to import `Glean` explicitly in test mode.
+@testable import Glean
+import XCTest
+
+class GleanUsageTests: XCTestCase {
+    override func setUp() {
+        Glean.shared.resetGlean(clearStores: true)
+    }
+
+    // ...
+}
+```
+
+This will ensure that metrics are done recording when the other test functions are used.
+
+</div>
+
+<div data-lang="Python" class="tab">
+
+The Glean Python SDK contains a helper function `glean.testing.reset_glean()` for resetting Glean for tests.
+It has two required arguments: the application ID, and the application version.
+
+Each reset of the Glean Python SDK will create a new temporary directory for Glean to store its data in.
+This temporary directory is automatically cleaned up the next time the Glean Python SDK is reset or when the testing framework finishes.
+
+The instructions below assume you are using [pytest](https://pypi.org/project/pytest/) as the test runner.
+Other test-running libraries have similar features, but are different in the details.
+
+Create a file `conftest.py` at the root of your test directory, and add the following to reset Glean at the start of every test in your suite:
+
+```python
+import pytest
+from glean import testing
+
+@pytest.fixture(name="reset_glean", scope="function", autouse=True)
+def fixture_reset_glean():
+    testing.reset_glean(application_id="my-app-id", application_version="0.1.0")
+```
+
+</div>
+
+<div data-lang="Rust" class="tab"></div>
+<div data-lang="JavaScript" class="tab">
+
+The Glean JavaScript SDK contains a helper function `testResetGlean()` for resetting Glean for tests.
+It expects the same list of arguments as `Glean.initialize`.
+
+Each reset of the Glean JavaScript SDK will clear stores. Calling `testResetGlean` will also make
+metrics and pings testing APIs available and replace ping uploading with a mock implementation that
+does not make real HTTP requests.
+
+```js
+import { testResetGlean } from "@mozilla/glean/testing"
+
+describe("myTestSuite", () => {
+    beforeEach(async () => {
+        await testResetGlean("my-test-id");
+    });
+});
+
+```
+
+</div>
+<div data-lang="Firefox Desktop" class="tab"></div>
```

Comment:
@chutten I am assuming testing Glean metrics in Firefox Desktop means Glean resetting is handled internally, is that correct? Would a message such as "On Firefox Desktop Glean resetting is handled internally." be correct here?

Reply:
Accccctttually, we just implemented `Services.fog.testResetFog()` for test reset capability (cf [bug 1741674](https://bugzilla.mozilla.org/show_bug.cgi?id=1741674)). You can read all about it in [our brand new instrumentation testing docs](https://firefox-source-docs.mozilla.org/toolkit/components/glean/user/instrumentation_tests.html).

What I missed was linking heavily to that doc from the `Firefox Desktop` tabs in all the metric types. Want me to file a bug for that?

Reply:
Nah, we do not mention resetting for any of the SDKs on the metric types pages. Actually this PR is the first time resetting is documented in the reference part of the book. Before it was mentioned in the "testing metric" guide only.

Reply:
Back to "what I should add here", maybe the tooltip can say "Resetting in FOG is documented on <this link you just sent me>"? WDYT?

Reply:
Sounds good to me.